### PR TITLE
ci: updated for best practices

### LINKED ISSUE #2243: update ci workflows for best practices and standards
we need to update the Github Actions CI pipelines to follow industry and organization best practices. This includes but is not limited to using commit locked versions of third-party actions, using self-hosted runners with advanced security, and adding explicit permissions to each workflow.

### PR DESCRIPTION
**Pull request by rbarkerSL**
**Description**:
Updating workflows as part of the CI audit on hashgraph org repos. 

**Related issue(s)**:

Fixes #2243

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/react_native.yml [hunk 1]
```diff
@@ -13,12 +13,17 @@ on:
...
+    run:
+        shell: bash
+
 permissions:
   contents: read
 
 jobs:
   android:
     name: Android
+    # runs-on: [ self-hosted, Linux, medium, ephemeral ]  # this will be enabled once we configure a self-hosted runner
```
Comment (rbarkerSL) [inline .github/workflows/react_native.yml:26]: Need to talk about macos-latest and iOS self-hosted runners

**Review by Svetoslav11 — APPROVED** [review]

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,2 +1,30 @@
...
+
+#########################
+#####  Core Files  ######
+#########################
+
+# NOTE: Must be placed last to ensure enforcement over all other rules
+
+# Protection Rules for Github Configuration Files and Actions Workflows
+/.github/                                       @hashgraph/release-engineering @hashgraph/release-engineering-managers
+/.github/workflows/                             @hashgraph/release-engineering @hashgraph/release-engineering-managers @SimiHunjan @ochikov @petreze @svetoslav-nikol0v @agadzhalov
```
Comment (agadzhalov) [inline .github/CODEOWNERS:12]: **Suggestion**: to create a group for the maintainers as in [C++](https://github.com/hashgraph/hedera-sdk-cpp/blob/main/.github/CODEOWNERS)

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,2 +1,30 @@
...
 *   @SimiHunjan @ochikov @petreze @svetoslav-nikol0v @agadzhalov
+
+#########################
+#####  Core Files  ######
+#########################
+
+# NOTE: Must be placed last to ensure enforcement over all other rules
+
+# Protection Rules for Github Configuration Files and Actions Workflows
+/.github/                                       @hashgraph/release-engineering @hashgraph/release-engineering-managers
```
Comment (agadzhalov) [inline .github/CODEOWNERS:11]: why maintainers should not have access to `.github` folder?

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -1,2 +1,30 @@
...
+/.github/workflows/                             @hashgraph/release-engineering @hashgraph/release-engineering-managers @hashgraph/hedera-sdk @hashgraph/hedera-sdk-js-contributors @SimiHunjan
```
Comment (agadzhalov) [inline .github/CODEOWNERS:12]: [@hashgraph/hedera-sdk-js-contributors](https://github.com/orgs/hashgraph/teams/hedera-sdk-js-contributors) in this group there are people who are not actively working on the JS SDK right now and on every new PR they are going to be spammed with emails. 
1. I think we should drop the contributors group. 
2. Second thing is in [hedera-sdk](https://github.com/orgs/hashgraph/teams/hedera-sdk) only Robert is a member, and he is not working on the JS SDK. 
3. In my opinion we should have a group similar to `hedera-sdk-java-maintainers` but `hedera-sdk-js-maintainers` and in this group to be only the people working actively on the JS SDK right now, and only they to be receive emails for new PRs

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -1,2 +1,30 @@
...
+# Protection Rules for Github Configuration Files and Actions Workflows
+/.github/                                       @hashgraph/release-engineering @hashgraph/release-engineering-managers
+/.github/workflows/                             @hashgraph/release-engineering @hashgraph/release-engineering-managers @hashgraph/hedera-sdk @hashgraph/hedera-sdk-js-maintainers
+
+# Codacy Tool Configurations
+/config/                                        @hashgraph/release-engineering @hashgraph/release-engineering-managers @hashgraph/hedera-sdk @hashgraph/hedera-sdk-js-maintainers
+.remarkrc                                       @hashgraph/release-engineering @hashgraph/release-engineering-managers @hashgraph/hedera-sdk @hashgraph/hedera-sdk-js-maintainers
+
+# Self-protection for root CODEOWNERS files (this file should not exist and should definitely require approval)
+/CODEOWNERS                                     @hashgraph/release-engineering @hashgraph/release-engineering-managers @hashgraph/hedera-sdk @hashgraph/hedera-sdk-js-maintainers
```
Comment (nathanklick) [inline .github/CODEOWNERS:19]: This line is prevent abberrent creation of this file and therefore should be scoped to our team. 

```suggestion
/CODEOWNERS                                     @hashgraph/release-engineering @hashgraph/release-engineering-managers
```

**Review by nathanklick — APPROVED** [review]

**Pull request merged**